### PR TITLE
#CX-12187 : Allow hover hover behavior to toggle menu overlay.

### PR DESCRIPTION
**Pull request by vikcool2pr**
## Description
Menu-overlay component allows toggling overlay on click by trigger element.
Same behavior is implemented with hovering trigger element with this implementation.

## Screenshots:
**Before** (If applicable):

**After**:

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Documentation update
- [ ] Code style update (formatting, local variables)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Other... Please describe:

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by namagoel — APPROVED** [review]

**Review by Samreen26 — APPROVED** [review]

**Review by prakhar-tiwari — APPROVED** [review]

File: web-components/src/components/menu-overlay/MenuOverlay.ts [hunk 1]
```diff
@@ -76,6 +77,7 @@ export namespace MenuOverlay {
     @property({ type: Boolean, attribute: "show-arrow" }) showArrow = false;
     @property({ type: Boolean }) disabled = false;
     @property({ type: String }) placement: MenuOverlay.Placement = "bottom";
+    @property({ type: Boolean, attribute: "allow-hover-toggle" }) allowHoverToggle = false;
```
Comment (momeraj) [inline web-components/src/components/menu-overlay/MenuOverlay.ts:80]: I feel the name of the attribute is not conveying the purpose "allow-hover-toggle".
can we name as "triggeredByHover"?

**Review by momeraj — CHANGES_REQUESTED** [review]

File: web-components/src/[sandbox]/examples/menu-overlay.ts [hunk 2]
```diff
@@ -331,7 +331,10 @@ export const menuOverlayTemplate = html`
   
 
   <h3 class="sandbox-header" style="margin: .5rem 1rem">with list (focus)</h3>
-  <md-menu-overlay class="queueDropdown" size="large" @menu-overlay-open=${(e: any) => {console.log("Opening modal--");document.dispatchEvent(new CustomEvent("on-widget-update"))}}>
+  <md-menu-overlay class="queueDropdown" size="large" @menu-overlay-open=${(e: any) => {
+    console.log("Opening modal--");
```
Comment (parsoman) [inline web-components/src/[sandbox]/examples/menu-overlay.ts:335]: Remove console logs

Comment (vikcool2pr) [reply web-components/src/[sandbox]/examples/menu-overlay.ts:335]: this log is from example.

File: web-components/src/components/menu-overlay/MenuOverlay.ts (hunk 1 above)
Comment (vikcool2pr) [reply web-components/src/components/menu-overlay/MenuOverlay.ts:80]: triggeredByHover looks a internal component level event. here this is kind of allowing or not flag. which looks fine to me.

**Review by momeraj — APPROVED** [review]

**Pull request merged**